simplefi-mstable-v2/src/feederPool.ts

```diff
@@ -0,0 +1,569 @@
+  redeemFromMarket,
+  TokenBalance,
+  updateMarket
+} from "./common"
+
+
+function getOrCreateLiquidity(feederPool: FeederPoolEntity, accountAddress: Address): AccountLiquidityEntity {
+  let id = feederPool.id.concat("-").concat(accountAddress.toHexString())
+  let liqudity = AccountLiquidityEntity.load(id)
+  if (liqudity != null) {
+    return liqudity as AccountLiquidityEntity
+  }
+  liqudity = new AccountLiquidityEntity(id)
+  liqudity.mAsset = feederPool.id
+  liqudity.account = getOrCreateAccount(accountAddress).id
+  liqudity.balance = BigInt.fromI32(0)
+  liqudity.save()
+  return liqudity as AccountLiquidityEntity
+}
+
+function updateMAsset(
+  event: ethereum.Event,
+  feederPool: FeederPoolEntity,
+  mAssetBalance: BigInt,
+  fAssetBalance: BigInt,
+  totalSupply: BigInt
+): FeederPoolEntity {
+  feederPool.mAssetBalance = mAssetBalance
+  feederPool.fAssetBalance = fAssetBalance
+  feederPool.totalSupply = totalSupply
+  feederPool.save()
+
+  let market = MarketEntity.load(feederPool.id) as MarketEntity
+
+  let inputTokenBalances: TokenBalance[] = []
+  inputTokenBalances.push(new TokenBalance(feederPool.mAsset, feederPool.id, mAssetBalance))
+  inputTokenBalances.push(new TokenBalance(feederPool.fAsset, feederPool.id, fAssetBalance))
+
+  updateMarket(
+    event,
+    market,
+    inputTokenBalances,
+    totalSupply
+  )
+
+  return feederPool
+}
+
+function handleMint(
+  event: ethereum.Event,
+  feederPool: FeederPoolEntity,
+  minter: string,
+  recipient: Address,
+  fpAssetQuantity: BigInt,
+  mAssetQuantity: BigInt,
+  fAssetQuantity: BigInt
+): void {
+  // update mAsset and market
+  let oldTotalSupply = feederPool.totalSupply
+  let newTotalSupply = oldTotalSupply.plus(fpAssetQuantity)
+
+  let newMAssetBalance = feederPool.mAssetBalance.plus(mAssetQuantity)
+  let newFAssetBalance = feederPool.fAssetBalance.plus(fAssetQuantity)
+
+  feederPool = updateMAsset(
+    event,
+    feederPool,
+    newMAssetBalance,
+    newFAssetBalance,
+    newTotalSupply
+  )
+
+  let account = getOrCreateAccount(recipient)
+
+  // Update account liquidity
+  let accountLiquidity = getOrCreateLiquidity(feederPool, recipient)
+  accountLiquidity.balance = accountLiquidity.balance.plus(fpAssetQuantity)
+  accountLiquidity.save()
+
+  let inputTokenBalances: TokenBalance[] = []
+  let amAssetBalance = newMAssetBalance
+  let afAssetBalance = newFAssetBalance
+  if (feederPool.totalSupply != BigInt.fromI32(0)) {
+    amAssetBalance = newMAssetBalance.times(accountLiquidity.balance).div(feederPool.totalSupply)
+    afAssetBalance = newFAssetBalance.times(accountLiquidity.balance).div(feederPool.totalSupply)
+  }
+  inputTokenBalances.push(new TokenBalance(feederPool.mAsset, account.id, amAssetBalance))
+  inputTokenBalances.push(new TokenBalance(feederPool.fAsset, account.id, afAssetBalance))
+
+  let inputTokenAmounts: TokenBalance[] = []
+  inputTokenBalances.push(new TokenBalance(feederPool.mAsset, minter, mAssetQuantity))
```

Comment:
Should be pushed to `inputTokenAmounts` instead of `inputTokenBalances` ?

Reply:
done